packages/sanity/src/core/studio/components/navbar/free-trial/FreeTrialContext.tsx

```diff
@@ -0,0 +1,75 @@
+import React, {createContext, useContext, useState, useCallback, useEffect} from 'react'
+import {useClient} from '../../../../hooks'
+import {SANITY_VERSION} from '../../../../version'
+import {FreeTrialResponse} from './types'
+
+interface FreeTrialContextProps {
+  data: FreeTrialResponse | null
+  showDialog: boolean
+  showOnLoad: boolean
+  /**
+   * If the user is seeing the `showOnLoad` popover or modal, and clicks on the pricing button the `showOnClick` modal should be triggered.
+   */
+  toggleShowContent: (closeAndReOpen?: boolean) => void
+}
+
+const FreeTrialContext = createContext<FreeTrialContextProps | undefined>(undefined)
+
+interface FreeTrialProviderProps {
+  children: React.ReactNode
+}
+export const FreeTrialProvider = ({children}: FreeTrialProviderProps) => {
+  const [data, setData] = useState<FreeTrialResponse | null>(null)
+  const [showDialog, setShowDialog] = useState(false)
+  const [showOnLoad, setShowOnLoad] = useState(false)
+  const client = useClient({apiVersion: 'vX'})
+
+  const fetchData = async () => {
+    const response = await client.request<FreeTrialResponse | null>({
+      url: `/journey/trial?studioVersion=${SANITY_VERSION}`,
+    })
+
+    setData(response)
+    // Validates if the user has seen the "structure rename modal" before showing this one. To avoid multiple popovers at same time.
+    const deskRenameSeen = localStorage.getItem('sanityStudio:desk:renameDismissed') === '1'
+    if (deskRenameSeen && response?.showOnLoad) {
+      setShowOnLoad(true)
+      setShowDialog(true)
+    }
+  }
+
+  useEffect(() => {
+    fetchData()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Just a thought: instead of forcing an empty dependency array, would it make sense to move the `fetchData` function definition _inside_ of the effect? It doesn't seem to be used elsewhere, and I think its only dependency is `client`, which should be stable.